### PR TITLE
minor refactors on plugins/modules/cloud/misc

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Minor refactors on modules under `plugins/modules/cloud/misc`

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugins/modules/cloud/misc/cloud_init_data_facts.py
plugins/modules/cloud/misc/proxmox_group_info.py
plugins/modules/cloud/misc/proxmox_kvm.py
plugins/modules/cloud/misc/rhevm.py
plugins/modules/cloud/misc/serverless.py
plugins/modules/cloud/misc/terraform.py

### CODE REVIEW

File: plugins/modules/cloud/misc/proxmox_kvm.py [hunk 1]
```diff
@@ -915,8 +912,9 @@ def create_vm(module, proxmox, vmid, newid, node, name, memory, cpu, cores, sock
         if 'pool' in kwargs:
             del kwargs['pool']
 
-    # Convert all dict in kwargs to elements. For hostpci[n], ide[n], net[n], numa[n], parallel[n], sata[n], scsi[n], serial[n], virtio[n], ipconfig[n]
-    for k in list(kwargs.keys()):
+    # Convert all dict in kwargs to elements.
+    # For hostpci[n], ide[n], net[n], numa[n], parallel[n], sata[n], scsi[n], serial[n], virtio[n], ipconfig[n]
+    for k in kwargs.keys():
```
Comment (felixfontein) [inline plugins/modules/cloud/misc/proxmox_kvm.py:917]: This is a breaking change. `kwargs` is modified during iteration.

File: plugins/modules/cloud/misc/rhevm.py [hunk 2]
```diff
@@ -547,7 +547,7 @@ def set_Memory(self, name, memory):
 
     def set_Memory_Policy(self, name, memory_policy):
         VM = self.get_VM(name)
-        VM.memory_policy.guaranteed = int(int(memory_policy) * 1024 * 1024 * 1024)
+        VM.memory_policy.guaranteed = (int(memory_policy) * 1024 * 1024 * 1024)
```
Comment (felixfontein) [inline plugins/modules/cloud/misc/rhevm.py:550]:
```suggestion
        VM.memory_policy.guaranteed = int(memory_policy) * 1024 * 1024 * 1024
```

File: plugins/modules/cloud/misc/terraform.py [hunk 3]
```diff
@@ -227,7 +227,7 @@ def get_version(bin_path):
 
 
 def preflight_validation(bin_path, project_path, version, variables_args=None, plan_file=None):
-    if project_path in [None, ''] or '/' not in project_path:
+    if '/' not in project_path:
```
Comment (felixfontein) [inline plugins/modules/cloud/misc/terraform.py:230]:
```suggestion
    if project_path is None or '/' not in project_path:
```

File: plugins/modules/cloud/misc/proxmox_kvm.py (hunk 1 above)
Comment (russoz) [reply plugins/modules/cloud/misc/proxmox_kvm.py]: oopsie, thanks for catching.

**Review by felixfontein — APPROVED** [review]
I'll keep this open for some days so maintainers have a chance to take a look at this as well.

**Comment by felixfontein** [comment]
@russoz thanks for improving the modules!

**Pull request merged**